platform/base/core/src/main/java/com/peregrine/sitemap/PageRecognizerBase.java

```diff
@@ -37,17 +37,28 @@
     private final String pagePrimaryType;
     private final String pageContentPrimaryType;
     private final String excludeFromSiteMapPropertyName;
+    private final String excludeTreeFromSiteMapPropertyName;
 
     protected PageRecognizerBase(
             final String pagePrimaryType,
             final String pageContentPrimaryType,
-            final String excludeFromSiteMapPropertyName) {
+            final String excludeFromSiteMapPropertyName,
+            final String excludeTreeFromSiteMapPropertyName) {
         this.pagePrimaryType = pagePrimaryType;
         this.pageContentPrimaryType = pageContentPrimaryType;
         this.excludeFromSiteMapPropertyName = excludeFromSiteMapPropertyName;
+        this.excludeTreeFromSiteMapPropertyName = excludeTreeFromSiteMapPropertyName;
     }
 
     public final boolean isPage(final Page candidate) {
+        return isAPageBasically(candidate) && !isExcludedByProperty(excludeFromSiteMapPropertyName, candidate);
+    }
+
+    public final boolean isBucket(final Page candidate) {
+        return isAPageBasically(candidate) && !isExcludedByProperty(excludeTreeFromSiteMapPropertyName, candidate);
+    }
+
+    private boolean isAPageBasically(final Page candidate) {
```

Comment:
I struggle with this method name - not sure what **isAPageBasically** means by reading that name

Reply:
I struggled with finding a good name indeed. I've come up with an alternative `hasAllPageMarkers`, hope will be good enough.